app/probe.go

```diff
@@ -0,0 +1,102 @@
+package app
+
+import (
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/arquivei/foundationkit/errors"
+)
+
+// Probe stores the state of a probe (`true` or `false` for ok and not ok respectively).
+type Probe struct {
+	ok *bool
+}
+
+// Set changes the state of a probe. Use `true` for ok and `false` for not ok.
+func (p *Probe) Set(ok bool) {
+	*p.ok = ok
+}
+
+// SetOk sets the probe as ok. Same as `Set(true)`.
+func (p *Probe) SetOk() {
+	*p.ok = true
+}
+
+// SetNotOk sets the probe as not ok. Same as `Set(false)`.
+func (p *Probe) SetNotOk() {
+	*p.ok = false
+}
+
+// IsOk returns the state of the probe  (`true` or `false` for ok and not ok respectively).
+func (p *Probe) IsOk() bool {
+	return *p.ok
+}
+
+// ProbeGroup aggregates and manages probes.
+type ProbeGroup struct {
+	lock   *sync.RWMutex
+	probes map[string]*bool
+}
+
+// NewProbeGroup returns a new ProbeGroup.
+func NewProbeGroup() ProbeGroup {
+	return ProbeGroup{
+		lock:   &sync.RWMutex{},
+		probes: make(map[string]*bool),
+	}
+}
+
+// NewProbe returns a new Probe with the given name.
+func (m *ProbeGroup) NewProbe(name string, ok bool) (Probe, error) {
+	if _, ok := m.probes[name]; ok {
+		return Probe{}, errors.Errorf("probe '%s' already registered", name)
+	}
+
+	if err := m.checkName(name); err != nil {
+		return Probe{}, err
+	}
+
+	s := ok
+	m.probes[name] = &s
+	return Probe{&s}, nil
```

Comment:
If the map is of probes, and you want to return a modifiable probe, you can return the address of the probe. Escape analysis will do the trick =D. I'd have a function that return the probes by name, too. It's cheap and might add some flexibility to the programmer.

Reply:
> If the map is of probes, and you want to return a modifiable probe, you can return the address of the probe.
I was trying to make is simple by making a map of bools, but maybe that was not a good call.

> I'd have a function that return the probes by name, too.
I didn't implement this one on purpose. My idea was that probes are managed by a single piece of code, so it would request a new probe for itself and use it.
Managing the same probe from different parts of the code could lead to confusion on why a certain probe is not ok.